### PR TITLE
feat(NIKKE.GG): add new presence

### PR DESCRIPTION
**Pull request by sinkaroid**
## Description 
This PR add NIKKE.GG (https://nikke.gg/) Presence
Summary: Nikke: Goddess of Victory Database, Wiki, Guides

> The website using wordpress, links and pathname probably a bit arbitrary, kindly review it

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
Screenshot_300  

## character
Screenshot_301
Screenshot_296

## monster
Screenshot_302
Screenshot_299

## tierlist
Screenshot_295

## guide
Screenshot_297
Screenshot_298

</details>

### CODE REVIEW

File: websites/N/NIKKE.GG/presence.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+			presenceData.smallImageText = "Reading";
+			presenceData.buttons = [{ label: "View Rapture", url: href }];
+		}
+	} else if (pathname.startsWith("/tier-list")) {
+		presenceData.largeImageKey = thumbnail || Assets.Logo;
+		presenceData.state = `${title.split("|")[0]}`;
+		presenceData.details = "Viewing Tier List";
+		presenceData.smallImageKey = Assets.Reading;
+		presenceData.smallImageText = "Reading";
+		presenceData.buttons = [{ label: "View Tierlist", url: href }];
```
Comment (theusaf) [inline websites/N/NIKKE.GG/presence.ts:78]:
```suggestion
		presenceData.buttons = [{ label: "View Tier List", url: href }];
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/N/NIKKE.GG/presence.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+			presenceData.largeImageKey = getImageCharacter(thumbnail);
+			presenceData.state = `${title.split("|")[0]}`;
+			presenceData.details = "Viewing Character";
+			presenceData.smallImageKey = Assets.Reading;
+			presenceData.smallImageText = "Reading";
+			presenceData.buttons = [{ label: "View Character", url: href }];
+		}
+	} else if (pathname.startsWith("/monsters")) {
+		if (pathname === "/monsters/") {
+			presenceData.largeImageKey = Assets.Logo;
```
Comment (DarkVillager) [inline websites/N/NIKKE.GG/presence.ts:51]: ?

File: websites/N/NIKKE.GG/presence.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+		{ href, pathname, search } = document.location,
+		thumbnail = document
+			.querySelector<HTMLMetaElement>('meta[property="og:image"]')
+			?.getAttribute("content"),
+		title = document
+			.querySelector<HTMLMetaElement>('meta[property="og:title"]')
+			?.getAttribute("content");
+
+	if (search.startsWith("?s")) {
+		presenceData.largeImageKey = Assets.Logo;
```
Comment (DarkVillager) [inline websites/N/NIKKE.GG/presence.ts:32]: ?

File: websites/N/NIKKE.GG/presence.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+		presenceData.state = search.split("&")[0].split("=")[1].replace(/\+/g, " ");
+		presenceData.details = "Searching for";
+		presenceData.smallImageKey = Assets.Search;
+	} else if (pathname.startsWith("/characters")) {
+		if (pathname === "/characters/") {
+			presenceData.largeImageKey = Assets.Logo;
```
Comment (DarkVillager) [inline websites/N/NIKKE.GG/presence.ts:38]:
```suggestion
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/N/NIKKE.GG/presence.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+		presenceData.buttons = [{ label: "View Tier List", url: href }];
+	} else if (pathname.startsWith("/") && pathname.length > 1) {
+		presenceData.largeImageKey = thumbnail || Assets.Logo;
+		presenceData.state = `${title.split("|")[0]}`;
+		presenceData.details = "Viewing Guide";
+		presenceData.smallImageKey = Assets.Reading;
+		presenceData.smallImageText = "Reading";
+		presenceData.buttons = [{ label: "View Guide", url: href }];
+	}
+	if (presenceData.details) presence.setActivity(presenceData);
```
Comment (DarkVillager) [inline websites/N/NIKKE.GG/presence.ts:87]:
```suggestion
	if (!presenceData.smallImageKey) { 
		presenceData.smallImageKey = Assets.Reading;
		presenceData.smallImageText = "Reading";
	}
	if (presenceData.details) presence.setActivity(presenceData);
```

File: websites/N/NIKKE.GG/presence.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+		if (pathname === "/characters/") {
+			presenceData.largeImageKey = Assets.Logo;
+			presenceData.details = "Viewing Characters List";
+			presenceData.smallImageKey = Assets.Viewing;
+		} else {
+			presenceData.largeImageKey = getImageCharacter(thumbnail);
+			presenceData.state = `${title.split("|")[0]}`;
+			presenceData.details = "Viewing Character";
+			presenceData.smallImageKey = Assets.Reading;
+			presenceData.smallImageText = "Reading";
```
Comment (DarkVillager) [inline websites/N/NIKKE.GG/presence.ts:46]:
```suggestion
```

File: websites/N/NIKKE.GG/presence.ts [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+			)}`;
+			presenceData.state = `${
+				document
+					.querySelectorAll<HTMLMetaElement>('meta[property="og:title"]')[1]
+					.getAttribute("content")
+					.split("|")[0]
+			}`;
+			presenceData.details = "Viewing Rapture";
+			presenceData.smallImageKey = Assets.Reading;
+			presenceData.smallImageText = "Reading";
```
Comment (DarkVillager) [inline websites/N/NIKKE.GG/presence.ts:69]:
```suggestion
```

File: websites/N/NIKKE.GG/presence.ts [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+			presenceData.buttons = [{ label: "View Rapture", url: href }];
+		}
+	} else if (pathname.startsWith("/tier-list")) {
+		presenceData.largeImageKey = thumbnail || Assets.Logo;
+		presenceData.state = `${title.split("|")[0]}`;
+		presenceData.details = "Viewing Tier List";
+		presenceData.smallImageKey = Assets.Reading;
+		presenceData.smallImageText = "Reading";
```
Comment (DarkVillager) [inline websites/N/NIKKE.GG/presence.ts:77]:
```suggestion
```

File: websites/N/NIKKE.GG/presence.ts [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+		presenceData.buttons = [{ label: "View Tier List", url: href }];
+	} else if (pathname.startsWith("/") && pathname.length > 1) {
+		presenceData.largeImageKey = thumbnail || Assets.Logo;
+		presenceData.state = `${title.split("|")[0]}`;
+		presenceData.details = "Viewing Guide";
+		presenceData.smallImageKey = Assets.Reading;
+		presenceData.smallImageText = "Reading";
```
Comment (DarkVillager) [inline websites/N/NIKKE.GG/presence.ts:84]:
```suggestion
```

File: websites/N/NIKKE.GG/presence.ts [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+				document
+					.querySelectorAll<HTMLMetaElement>('meta[property="og:image"]')[1]
+					.getAttribute("content")
+					.replace(".png.png", ".png")
+			)}`;
+			presenceData.state = `${
+				document
+					.querySelectorAll<HTMLMetaElement>('meta[property="og:title"]')[1]
+					.getAttribute("content")
+					.split("|")[0]
```
Comment (DarkVillager) [inline websites/N/NIKKE.GG/presence.ts:65]:
```suggestion
				document.querySelector('article')?.querySelector("h1")?.textContent
```

File: websites/N/NIKKE.GG/presence.ts [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+			presenceData.largeImageKey = Assets.Logo;
+			presenceData.details = "Viewing Rapture List";
+			presenceData.smallImageKey = Assets.Viewing;
+		} else {
+			presenceData.largeImageKey = `${getImageMonster(
+				document
+					.querySelectorAll<HTMLMetaElement>('meta[property="og:image"]')[1]
+					.getAttribute("content")
+					.replace(".png.png", ".png")
+			)}`;
```
Comment (DarkVillager) [inline websites/N/NIKKE.GG/presence.ts:60]: This is simpler and works
```suggestion
			presenceData.largeImageKey = document.querySelector<HTMLImageElement>('.img > img')?.src ?? Assets.Logo
```

File: websites/N/NIKKE.GG/presence.ts [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+			startTimestamp: browsingTimestamp,
+		},
+		getImageCharacter = (url: string) =>
+			`${"https://static.dotgg.gg/nikke/characters/"}si_c${
+				url.match(/c(\d+)_\d+/)[1]
+			}_${url.match(/c\d+_(\d+)/)[1]}_s.webp`,
+		getImageMonster = (url: string) =>
+			`https://static.dotgg.gg/nikke/monsters/si_${url
+				.substring(url.lastIndexOf("/") + 1, url.lastIndexOf("."))
+				.replace("full_", "")}.webp`,
```
Comment (DarkVillager) [inline websites/N/NIKKE.GG/presence.ts:22]:
```suggestion
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]
If you have any questions about any of the requested changes feel free to ask in the discord server or in my dms on discord.

File: websites/N/NIKKE.GG/presence.ts (hunk 3 above)
Comment (DarkVillager) [inline websites/N/NIKKE.GG/presence.ts:32]:
```suggestion
```

File: websites/N/NIKKE.GG/presence.ts (hunk 2 above)
Comment (DarkVillager) [inline websites/N/NIKKE.GG/presence.ts:51]:
```suggestion
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/N/NIKKE.GG/presence.ts (hunk 10 above)
Comment (sinkaroid) [reply websites/N/NIKKE.GG/presence.ts]: Adjusted

File: websites/N/NIKKE.GG/presence.ts (hunk 11 above)
Comment (sinkaroid) [reply websites/N/NIKKE.GG/presence.ts:55]: That directly images not resizing properly for a thumbnail (need at least symmetrical), that's why came from another but still had relation

**Review by Bas950 — APPROVED** [review]

**Review by Timeraa — APPROVED** [review]

File: websites/N/NIKKE.GG/presence.ts (hunk 1 above)
Comment (sinkaroid) [reply websites/N/NIKKE.GG/presence.ts]: Changes updated

**Pull request merged**